src/middlewared/middlewared/validators.py

```diff
@@ -34,6 +35,25 @@ def __call__(self, value):
             raise ShouldBe("valid IP address")
 
 
+class Time:
+    def __call__(self, value):
+        try:
+            hours, minutes = value.split(':')
+        except ValueError:
+            raise ShouldBe('Time should be in 24 hour format like "18:00"')
+        else:
+            time_dict = {'hours': hours, 'minutes': minutes}
+            try:
+                for key in time_dict.keys():
+                    if not time_dict[key].isdigit():
+                        raise TypeError
+                time(int(time_dict['hours']), int(time_dict['minutes']))
+            except TypeError:
+                raise ShouldBe(f'{key} should be a number. It should be like "18:00"')
+            except ValueError as v:
+                raise ShouldBe(f'{v.__str__()}')
```

Comment:
No need to abuse f-string neither magic methods. raise ShouldBe(str(v)) should do the job just fine